### PR TITLE
Update holocron - provide rejected modules

### PR DESCRIPTION
**Pull request by JAdshead**
## Description

Consuming the latest Holocron version which includes rejected module updates.
Makes use of rejected modules to provide additional logs and reset the module map poll time.

Additional PR should follow which adds the rejected module information in the metrics.

## Motivation and Context

After Holocron resiliency update one-app no longer reset when a module is rejected. The error is also logged only once which reduces the visibility of the error.  
The aim of this is to improve DX when modules rejected. 

## How Has This Been Tested?

Locally and test suite.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: src/server/utils/pollModuleMap.js [hunk 1]
```diff
@@ -23,11 +23,6 @@ import {
 
   holocron as holocronMetrics,
 } from '../metrics';
-import {
-  getModulesUsingExternals,
-  setModulesUsingExternals,
-} from './onModuleLoad';
-import { backupModuleStateConfig, restoreModuleStateConfig } from './stateConfig';
```
Comment (Matthew-Mallimo) [inline src/server/utils/pollModuleMap.js:30]: was this not being used?

File: __tests__/server/utils/pollModuleMap.spec.js [hunk 2]
```diff
@@ -350,22 +263,34 @@ describe('pollModuleMap', () => {
...
+    expect(console.warn).toHaveBeenCalledTimes(1);
+    expect(console.warn).toHaveBeenCalledWith('pollModuleMap: 1 modules rejected:', ['bad-module: not compatible']);
+
+    expect(setTimeout).toHaveBeenCalledTimes(1);
+    expect(setTimeout).toHaveBeenCalledWith(pollModuleMap, 5e3);
+
+    expect(setGauge).toHaveBeenCalledTimes(1);
+    expect(setGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollWait, 5);
+    expect(incrementGauge).toHaveBeenCalledTimes(1);
+    expect(incrementGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollConsecutiveErrors);
```
Comment (Matthew-Mallimo) [inline __tests__/server/utils/pollModuleMap.spec.js:293]: Can we also check that `resetPollTime` was called

File: __tests__/server/utils/pollModuleMap.spec.js [hunk 3]
```diff
@@ -350,22 +263,34 @@ describe('pollModuleMap', () => {
...
-
-        expect(setGauge).toHaveBeenCalledTimes(1);
-        expect(setGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollWait, 5);
-        expect(incrementGauge).toHaveBeenCalledTimes(1);
-        expect(incrementGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollConsecutiveErrors);
-      });
+    expect(setGauge).toHaveBeenCalledTimes(1);
+    expect(setGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollWait, 5);
+    expect(incrementGauge).toHaveBeenCalledTimes(1);
+    expect(incrementGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollConsecutiveErrors);
```
Comment (Matthew-Mallimo) [inline __tests__/server/utils/pollModuleMap.spec.js:275]: check that `resetPollTime` was called as well

File: src/server/utils/pollModuleMap.js (hunk 1 above)
Comment (JAdshead) [reply src/server/utils/pollModuleMap.js:30]: modules are either accepted or rejected, we don't have a situation anymore where we revert a partially loaded module map

**Review by Matthew-Mallimo — CHANGES_REQUESTED** [review]
comments

File: __tests__/server/utils/pollModuleMap.spec.js [hunk 4]
```diff
@@ -289,59 +212,49 @@ describe('pollModuleMap', () => {
...
+
+    expect(setTimeout).toHaveBeenCalledTimes(1);
+    expect(setTimeout.mock.calls[0][0]).toBe(pollModuleMap);
+    expect(setTimeout.mock.calls[0][1]).toBeGreaterThanOrEqual(5e3 * 1.25);
+
+    expect(setGauge).toHaveBeenCalledTimes(1);
+    expect(setGauge.mock.calls[0][0]).toBe(holocronMetrics.moduleMapPollWait);
+    expect(setGauge.mock.calls[0][1]).toBeGreaterThanOrEqual(5 * 1.25);
+    // ensure we're using seconds, not milliseconds
+    expect(setGauge.mock.calls[0][1]).toBeLessThan(8 * 1.25);
```
[the remark this answers is not in the document]
Comment (JAdshead) [reply __tests__/server/utils/pollModuleMap.spec.js:257]: this is what the setTimeout check is doing - https://github.com/americanexpress/one-app/pull/962/files#diff-82cec72196769195fc18f858b882683ecae08a9f5fc3b7d097fae4f60d33b404R251

File: __tests__/server/utils/pollModuleMap.spec.js (hunk 3 above)
Comment (JAdshead) [reply __tests__/server/utils/pollModuleMap.spec.js:275]: this is what the setTimeout check is doing - https://github.com/americanexpress/one-app/pull/962/files#diff-82cec72196769195fc18f858b882683ecae08a9f5fc3b7d097fae4f60d33b404R270

File: __tests__/server/utils/pollModuleMap.spec.js (hunk 2 above)
Comment (JAdshead) [reply __tests__/server/utils/pollModuleMap.spec.js:293]: https://github.com/americanexpress/one-app/pull/962/files#diff-82cec72196769195fc18f858b882683ecae08a9f5fc3b7d097fae4f60d33b404R288

File: __tests__/server/utils/pollModuleMap.spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Matthew-Mallimo) [reply __tests__/server/utils/pollModuleMap.spec.js:257]: In that case can we leave a comment above the line indicating thats what that line is testing? It's not immediately clear

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**